Danki2/Assets/Scripts/Actor/Services/MovementManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿using UnityEngine;
+
+public class MovementManager : MonoBehaviour
+{
+    float _moveLockRemainingDuration;
+    float _moveLockSpeed;
+    bool _moveLockOn;
+    Vector3 _moveLockDirection;
+    Vector3 _moveDirection;
+
+    void Start()
+    {
+        _moveLockRemainingDuration = 0f;
+        _moveLockSpeed = 0f;
+        _moveLockOn = false;
+        _moveLockDirection = Vector3.one;
+        _moveDirection = Vector3.zero;
+    }
+
+    void Update()
+    {
+        _moveLockRemainingDuration = Mathf.Max(0f, _moveLockRemainingDuration - Time.deltaTime);
+        _moveLockOn = _moveLockRemainingDuration > 0 ? _moveLockOn : false;
+    }
+
+    private void LateUpdate()
+    {
+        var speedStat = gameObject.GetComponent<StatsManager>()[Stat.Speed];
+
+        var movementVector = _moveLockOn
+            ? Vector3.Normalize(_moveLockDirection) * _moveLockSpeed
+            : Vector3.Normalize(_moveDirection) * speedStat;
+
+        transform.Translate(movementVector * Time.deltaTime);
+        _moveDirection = Vector3.zero;
+    }
+
+    public void LockMovement(float duration, float speed, Vector3 direction)
```

Comment:
May want to make the speed and direction nullable. So outside forces can lock our movement without having to know additional info about us. If a parameter is null it would just lock it to it's current value.

Reply:
The problem I can see with that is, you might lock movement before any movement for the frame has been registered? Besides, I can't think of an obvious use case for locking current movement.

Reply:
Yeah, fair enough.